chainer/functions/batch_normalization.py

```diff
@@ -130,6 +132,35 @@ def __call__(self, x, test=False, finetune=False):
         self.is_finetune = finetune
         return function.Function.__call__(self, x)
 
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+
+        if isinstance(self.size, tuple):
+            feature_shape = type_check.Variable(self.size, 'feature_shape')
```

Comment:
How about making `self` as a variable (e.g. named `self_`) instead of feature_shape?